### PR TITLE
Fix project service as per new audit log design

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

**Review by conradoverta — APPROVED** [review]

**Comment by conradoverta** [comment]
Please make sure it builds

File: backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java [hunk 1]
```diff
@@ -1278,10 +1355,16 @@ public void deleteProjects(
...
-          DeleteProjects.Response.newBuilder().setStatus(!deletedProjectIds.isEmpty()).build());
+      DeleteProjects.Response response =
+          DeleteProjects.Response.newBuilder().setStatus(!deletedProjectIds.isEmpty()).build();
+      saveAuditLogs(
+          Optional.empty(),
+          ModelDBServiceActions.DELETE,
+          deletedProjectIds,
+          ModelDBUtils.getStringFromProtoObject(request),
+          ModelDBUtils.getStringFromProtoObject(response),
+          authService.getWorkspaceIdFromUserInfo(authService.getCurrentLoginUserInfo()));
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: See https://github.com/VertaAI/modeldb/pull/1958#discussion_r576388000

**Review by conradoverta — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java [hunk 2]
```diff
@@ -1277,11 +1357,24 @@ public void deleteProjects(
...
-      saveAuditLogs(null, ModelDBConstants.DELETE, deletedProjectIds, "");
-
-      responseObserver.onNext(
-          DeleteProjects.Response.newBuilder().setStatus(!deletedProjectIds.isEmpty()).build());
+      DeleteProjects.Response response =
+          DeleteProjects.Response.newBuilder().setStatus(!deletedProjectIds.isEmpty()).build();
+      Long workspaceId =
+          responseItems.stream().findFirst().isPresent()
+              ? responseItems.stream().findFirst().get().getWorkspaceId()
+              : null;
```
Comment (evgeniilezhnin) [inline backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: that is only for the first of projects from the list, right? See https://github.com/VertaAI/modeldb/pull/1980, maybe it's better

Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: I don't understand what you mean

Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: I mean that deleted projects may have different workspace ids. Personal workspace can be used here?

Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: The new protos should resolve this, right?

Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: I think so. We decided to use multiple workspaces?

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: We have logging multiple resourceIds at once in single audit log entry. and we do not have provision for now to use endpoint with multiple workspaceId's. So workspaceId will always there.

Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: This change has multiple workspace ids: https://github.com/VertaAI/protos-all/pull/162/files Is it a mistake?

Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: Yeah, that's what I noticed too. `repeated AuditResource resource` means that each item can have its own workspace according to the protos, right? Happy to move the conversation to slack for faster iteration.

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: PR: https://github.com/VertaAI/modeldb/pull/2027 will supporting resource and workspace together in single entry.

Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectServiceImpl.java]: Resolved in slack

**Review by conradoverta — APPROVED** [review]
Approved with the follow-up from Slack

**Pull request merged**